internal/csi-common/utils.go

```diff
@@ -320,3 +320,74 @@ func IsBlockMultiNode(caps []*csi.VolumeCapability) (bool, bool) {
 
 	return isBlock, isMultiNode
 }
+
+// IsFileRWO checks if it is of type RWO and file mode, if it is return value
+// will be set to true.
+func IsFileRWO(caps []*csi.VolumeCapability) bool {
+	// rwoFile has been set if the volume is of file mode and if the capabilities are of RWO
+	// kind, ie SINGLE NODE but flexible to have one or more writers. This is also used as a validation in caller
+	// to preserve the backward compatibility we had with file mode RWO volumes.
+	var rwoFile bool
+
+	// to preserve backward compatibility we allow RWO filemode, ideally SINGLE_NODE_WRITER check is good enough,
+	// however more granular level check could help us in future, so keeping it here as an additional measure.
+	for _, cap := range caps {
+		if cap.AccessMode != nil {
+			if cap.GetMount() != nil {
+				switch cap.AccessMode.Mode { //nolint:exhaustive // only check what we want
+				case csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+					csi.VolumeCapability_AccessMode_SINGLE_NODE_MULTI_WRITER,
+					csi.VolumeCapability_AccessMode_SINGLE_NODE_SINGLE_WRITER:
+					rwoFile = true
+				}
+			}
+		}
+	}
+
+	return rwoFile
+}
+
+// IsReaderOnly check and set return value true only when the access mode is `READER ONLY` regardless of file
+// or block mode.
+func IsReaderOnly(caps []*csi.VolumeCapability) bool {
+	var roOnly bool
+	for _, cap := range caps {
+		if cap.AccessMode != nil {
+			switch cap.AccessMode.Mode { //nolint:exhaustive // only check what we want
+			case csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY,
+				csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY:
+				roOnly = true
```

Comment:
similar here, just return `true` here, and drop the `roOnly` variable

Reply:
Done